pkg/dispatcher/cache/event_handlers.go

```diff
@@ -235,3 +208,27 @@ func (dc *DispatcherCache) updateResourceBinding(oldObj, newObj interface{}) {
 	dc.deleteResourceBinding(oldRb)
 	dc.addResourceBinding(newRb)
 }
+
+func (dc *DispatcherCache) getResourceFromObjectReference(ref workv1alpha2.ObjectReference) (*unstructured.Unstructured, error) {
+	gv, err := schema.ParseGroupVersion(ref.APIVersion)
+	if err != nil {
+		klog.Errorf("Failed to parse GroupVersion <%s>, err: %v", ref.APIVersion, err)
+		return nil, err
+	}
+	mapping, err := dc.restMapper.RESTMapping(schema.GroupKind{
+		Group: gv.Group,
+		Kind:  ref.Kind,
+	})
+
+	if err != nil {
+		klog.Errorf("Failed to get resource mapping from reference <%s/%s>, err: %v")
+		return nil, err
+	}
+
+	resource, err := dc.dynamicClient.Resource(mapping.Resource).Namespace(ref.Namespace).Get(context.Background(), ref.Name, metav1.GetOptions{})
```

Comment:
Is there a way that we can get resource directly and avoid API request, seems it's heavy.